lighthouse-core/lib/csp-evaluator.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * @typedef Finding
+ * @property {number} type
+ * @property {string} description
+ * @property {number} severity Severity value 0-100 where 0 is the most severe.
+ * @property {string} directive The directive the finding applies to.
+ * @property {string|undefined} value Keyword if the finding applies to one.
+ */
+
+const log = require('lighthouse-logger');
+const i18n = require('../lib/i18n/i18n.js');
+const {
+  Parser,
+  Type,
+  Directive,
+  evaluateForFailure,
+  evaluateForSyntaxErrors,
+  evaluateForWarnings,
+} = require('../../third-party/csp-evaluator/optimized_binary.js');
+
+const UIStrings = {
+  /** Message shown when a CSP does not have a base-uri directive. Shown in a table with a list of other CSP vulnerabilities and suggestions. "CSP" stands for "Content Security Policy". "base-uri", "'none'", and "'self'" do not need to be translated. */
+  missingBaseUri: 'Missing base-uri allows the injection of base tags. ' +
+    'They can be used to set the base URL for all relative (script) ' +
+    'URLs to an attacker controlled domain. ' +
+    'Can you set it to \'none\' or \'self\'?',
+  /** Message shown when a CSP does not have a script-src directive. Shown in a table with a list of other CSP vulnerabilities and suggestions. "CSP" stands for "Content Security Policy". "script-src" does not need to be translated. */
+  missingScriptSrc: 'script-src directive is missing. ' +
+    'This can allow the execution of unsafe scripts.',
+  /** Message shown when a CSP does not have a script-src directive. Shown in a table with a list of other CSP vulnerabilities and suggestions. "CSP" stands for "Content Security Policy". "object-src" and "'none'" do not need to be translated. */
+  missingObjectSrc: 'Consider setting object-src to \'none\' to prevent the injection of ' +
```

Comment:
I can't recall if we should put technical terms in back-ticks so they don't get translated.  is that a thing? @brendankenny @exterkamp wdyt?

Reply:
> I can't recall if we should put technical terms in back-ticks so they don't get translated. is that a thing?

Almost alway yes. It's also likely things like "object-src" will be left alone by the translators, but just to be sure you might want to wrap all those as well (and having them as `<pre>` text in the report seems like it might be useful regardless?)

Reply:
> and having them as `<pre>` text in the report seems like it might be useful regardless?

Does that just mean surrounding some words with `<pre>...</pre>` in these strings?

Reply:
Sorry, I mean the backticks should convert to that for most strings in the report

Reply:
Looks like backticks don't work for table items
![Screen Shot 2021-03-02 at 12 30 49 PM](https://user-images.githubusercontent.com/6752989/109689268-2056e600-7b53-11eb-9e37-39bc92143376.png)